imaging/ml/ml_codelab/scripts/inference/Dockerfile

```diff
@@ -19,15 +19,14 @@ RUN mkdir -p /opt/inference_module/src && mkdir -p /opt/inference_module/bin
 ADD / /opt/inference_module/src/
 
 # Install dependencies.
-RUN pip install --upgrade pip && pip install --upgrade virtualenv && \
-    virtualenv /opt/inference_module/venv && \
```

Comment:
any reason why you remove the virtualenv?

As rare as possible bugs may be, it is helpful in completely eliminating bugs which relate to managing Python versions on the OS level

Reply:
>any reason why you remove the virtualenv

- it's increase size of image
- it makes structure inside of image unreasonably complicated 

virtualenv inside of docker image is equivalent of vm inside of vm
environment is static and isolated from OS it runned in,  virtualenv is useful inside of docker only if we run py2 and py3 at the same time inside of same container

Reply:
i think this is more subtle than the "vm inside of vm" comparison, here are some thoughts:
https://vsupalov.com/virtualenv-in-docker/

if it is not that much work can you revert this, the size increase and added complexity was not an issue in this case

Reply:
>When using virtualenv & co, you can make sure that a particular Python version 

we are using same version each build already

>You can completely eliminate bugs which relate to managing Python versions on the OS level

python in image used only for single process,nothing else is installing dependencies 

- we use single version of python in image
- we run single process of python
- we need only one dependency tree, where is no possible conflicts
so in this case only thing is provided by venv is isolation, which is done by docker

if my opinion is not valid:
https://nickjanetakis.com/blog/docker-tip-41-should-you-use-virtualenv-or-rvm-in-your-docker-images